### ISSUE TITLE
CSV import/export multi-column natural keys

### ISSUE THREAD

**Issue by lampwins** [nautobot/nautobot#3794]
As Nelly the Network Engineer and Ozzie the Operator, I want CSV import/export formats to break out separate columns for composite natural key references, so that I am not confused by the format and structure of complex natural keys when I expect to provide individual data points in their own columns.

As an example, device type is referenced by it's manufacturer name and model number.

So, in the case of the device import/export, the goal will be to have "manufacturer" and "model_number" as discrete fields, however having "device_type.manufacturer" and "device_type.model_number" in the interim is acceptable. The point is that the ultimate ask is that when the list of fields in scope in unique, the most simple column headers is preferred.

**Comment by bryanculver** [comment]
Instead of
```
# import.csv
interface
ams01-edge-01;AMS01;Ethernet1%2F1
``` 

Have
```
inteface__device__location,interface__device__name,interface__name
AMS01,ams01-edge-01,Ethernet1%2F1
```

This should not replace existing functionality (with a composite key or #3976) and you cannot mix the broken out representation (example above) with the composite key or UUID single field representation.

**Comment by bryanculver** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Nautobot-Core-Engineering-64a875e74842500d9e80e23e/issues/nautobot/nautobot/3794?planning-poker) with Zenhub @glennmatthews @gsnider2195 @HanlinMiao @jathanism @timizuoebideri1

**Comment by glennmatthews** [comment]
Implemented in #4274.
